model_compression_toolkit/gptq/pytorch/quantizer/base_pytorch_gptq_quantizer.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2023 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from abc import abstractmethod
+from typing import Union, Dict, List
+
+from torch import Tensor
+
+from model_compression_toolkit.core.common.logger import Logger
+from model_compression_toolkit.core.common.constants import FOUND_TORCH
+from model_compression_toolkit.gptq.common.gptq_constants import WEIGHTS_QUANTIZATION_PARAMS
+
+from model_compression_toolkit.quantizers_infrastructure import TrainableQuantizerWeightsConfig, \
+    TrainableQuantizerActivationConfig, BasePytorchTrainableQuantizer
+
+if FOUND_TORCH:
+
+    class BasePytorchGPTQTrainableQuantizer(BasePytorchTrainableQuantizer):
+        """
+        A base class for trainable Pytorch quantizer for GPTQ.
+        """
+
+        def __init__(self,
+                     quantization_config: Union[TrainableQuantizerWeightsConfig, TrainableQuantizerActivationConfig]):
+            """
+            Initializes BasePytorchGPTQTrainableQuantizer object.
+
+            Args:
+                quantization_config: quantizer config class contains all the information about a quantizer configuration.
+            """
+
+            super().__init__(quantization_config)
+
+        def update_layer_quantization_params(self, layer) -> (Dict[str, Tensor], Dict[str, Dict], Dict):
+            """
+            A Function to calculate the needed change in attributes in NodeQuantizationConfig after retraining.
+
+            Returns:
+                3 dictionaries describing the change in layer's weights, weights config, activation config
+                that changed during GPTQ retraining.
+                Keys must match NodeQuantizationConfig attributes
+
+            """
+            weights = {}
+            for weight, quantizer_vars, quantizer in layer.get_weights_vars():
+                weights.update({weight: quantizer(training=False, inputs=quantizer_vars)})
+
+            quant_config = {WEIGHTS_QUANTIZATION_PARAMS: self.get_quant_config()}
+
+            return weights, quant_config, {}
```

Comment:
Is it possible to remove the empty activation config?

Reply:
Currently no. This function is supposed to be implemented in any GPTQ quantizer to allow updating the layers parameters according to the training results.
In theory, we might want to also update activation-related parameters (such as activation threshold that we tried to add before but it didn't work well), so we need to return 3 dictionaries from this method.